packages/ember-headless-form/src/components/-private/field.ts

```diff
@@ -123,7 +123,8 @@ export default class HeadlessFormFieldComponent<
   }
 
   get value(): DATA[KEY] {
-    return this.args.data[this.args.name];
+    // when @mutableData is set, data is something we don't control, i.e. might require old-school get() to be on the safe side
```

Comment:
> i.e. might require old-school get() to be on the safe side

I'm worried this enables `.`-separated keys, tying us to some form of `get` forever -- could this be something the adapter layer does?

Reply:
Do you think `get()` will/should go away? Like even in a future with deprecated/removed CPs, it could still exist as a simple util for nested property access? So I am not that worried about that tbh...

Also, I remember we chatted about nested data, which would require just this, not? This is certainly not enough to make nested data officially supported, would require test cases etc, but at the end we would need `get()` here anyway, right?

Reply:
If we don't want to leak the semantics of nested property access here, then we could also add an `assert()` for the path (name) to not contain any dots? 

This way we could at least make nested access an explicit choice later, when we are confident about its use as _our_ public API.

Reply:
>  add an assert() for the path (name) to not contain any dots?

yeah, I'm good with that

> will/should go away?

yeah, I can't speak on behalf of the core team, as I haven't talked with them about this, but -- it's hard to type correctly (and presently has "wrong" types which are "good enough", for performance reasons -- types on nested keys do not say what you'd expect).
In the future, I imagine we'd handle this situation differently, which _is_ kind of like an adapter-like thing.
get does _weird_ and _unexpected_ things with proxying the returned value (iirc) -- it def feels like tech debt when debugging it.

>  it could still exist as a simple util for nested property access? 

for this use case specifically, the core teams have said they want to discourage get usage from ember, and instead defer to other "utility libraries", such as "lodash.get".

>  we chatted about nested data, which would require just this, not?

it would, but we also talked about requiring an adapter and flat data if nested input-data was needed, and I kinda like that explicitness, as native `FormData` can _only_ understand flat data structures (even with dots, the field names are still a single property) -- changeset, and some forms I've built use a notation convention for translating between nested properties / objects / arrays via `.dot.property.0.index.4.access` -- it's.... it required a lot of tests in validated-changeset to get it all working 😅

Reply:
> add an assert() for the path (name) to not contain any dots?

> yeah, I'm good with that

Did that now!